## Hypthesis testing assignment Q4

TeleCall uses 4 centers around the globe to process customer order forms. They audit a certain %  of the customer order forms. Any error in order form renders it defective and has to be reworked before processing.  The manager wants to check whether the defective %  varies by centre. Please analyze the data at 5% significance level and help the manager draw appropriate inferences

Minitab File: CustomerOrderForm.mtw


In [29]:
#Import the required libraries required
import pandas as pd
import scipy.stats as stats
import numpy as np

In [2]:
#Import the data
data = pd.read_csv('CustomerOrderForm.csv')
data.head()

Phillippines   Indonesia       Malta       India
0   Error Free  Error Free   Defective  Error Free
1   Error Free  Error Free  Error Free   Defective
2   Error Free   Defective   Defective  Error Free
3   Error Free  Error Free  Error Free  Error Free
4   Error Free  Error Free   Defective  Error Free

H0 (Null Hypothesis):  The defective % does not varies by centre.

H1(Alternative Hypothesis): The defective % varies by centre.

Given the categorical varibles, So using Chi-Squared contingency independence test

In [22]:
#Making a contingency table with the given values
data.Phillippines.value_counts()

Error Free    271
Defective      29
Name: Phillippines, dtype: int64

In [23]:
data.Indonesia.value_counts()

Error Free    267
Defective      33
Name: Indonesia, dtype: int64

In [24]:
data.Malta.value_counts()

Error Free    269
Defective      31
Name: Malta, dtype: int64

In [25]:
data.India.value_counts()

Error Free    280
Defective      20
Name: India, dtype: int64

In [27]:
table = np.array([[271,267,269,280],[29,33,31,20]])
table

array([[271, 267, 269, 280],
       [ 29,  33,  31,  20]])

In [28]:
#Using chi2_contingency test
chi, pval, DOF, expected = stats.chi2_contingency(table)
pval

0.2771020991233135

Conclusion: 

p value(0.28) > (α = 0.05)

Hence failed to reject the null hypothesis. 

Independence of categorical variables Thus, customer order forms defective % does not varies by centre.